src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -26,6 +25,11 @@ public abstract class FileSystemWatcherTest : FileCleanupTestBase
         public const int DefaultAttemptsForExpectedEvent = 3;       // Number of times an expected event should be retried if failing.
         public const int DefaultAttemptsForUnExpectedEvent = 2;     // Number of times an unexpected event should be retried if failing.
 
+        public static FileSystemEventHandler changeHandler;
```

Comment:
You can't do this and have tests run concurrently. One possible answer is to have the `Watch*` methods return a tuple `(AutoResetEvent event, FileSystemEventHandler handler)` and unhook in the caller.

Even if this assembly is set to run sequentially, it would be bad to leave an avoidable time-bomb like this.